examples/api/lib/material/navigation_drawer/navigation_drawer.0.dart

```diff
@@ -0,0 +1,171 @@
+      debugShowCheckedModeBanner: false,
+      title: 'Material 3',
+      theme: ThemeData(useMaterial3: true),
+      home: LayoutBuilder(builder: (BuildContext context, BoxConstraints constraints) {
+        if (constraints.maxWidth < narrowScreenWidthThreshold) {
+          return Scaffold(
+            body: Center(
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                children: <Widget>[
+                  Text('Page Index =  $screenIndex'),
+                  ElevatedButton(
+                    onPressed: _openDrawer,
+                    child: const Text('Open Drawer'),
+                  ),
+                ],
+              ),
+            ),
+            bottomNavigationBar: NavigationBar(
+              selectedIndex: screenIndex,
+              onDestinationSelected: (int index) {
+                setState(() {
+                  screenIndex = index;
+                });
+              },
+              destinations: navigationDestinationDatas
+                  .map((NavigationDestinationData e) => e.buildNavigationBarDestination(context))
+                  .toList(),
+            ),
+          );
+        } else {
+          return Scaffold(
+            key: _scaffoldKey,
+            body: SafeArea(
+              bottom: false,
+              top: false,
+              child: Row(
+                children: <Widget>[
+                  Padding(
+                    padding: const EdgeInsets.symmetric(horizontal: 5),
+                    child: NavigationRail(
+                      minWidth: 50,
+                      destinations: navigationDestinationDatas
+                          .map((NavigationDestinationData data) => data.buildnRailDestination(context))
+                          .toList(),
+                      selectedIndex: screenIndex,
+                      useIndicator: true,
+                      onDestinationSelected: (int index) {
+                        setState(() {
+                          screenIndex = index;
+                        });
+                      },
+                    ),
+                  ),
+                  const VerticalDivider(thickness: 1, width: 1),
+                  Expanded(
+                    child: Column(
+                      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                      children: <Widget>[
+                        Text('Page Index =  $screenIndex'),
+                        ElevatedButton(
+                          onPressed: _openDrawer,
+                          child: const Text('Open Drawer'),
+                        ),
+                      ],
+                    ),
+                  ),
+                ],
+              ),
+            ),
+            endDrawer: myDrawer,
+          );
+        }
+      }),
+    );
+  }
+}
+
+const double narrowScreenWidthThreshold = 450;
+const List<NavigationDestinationData> navigationDestinationDatas = <NavigationDestinationData>[
```

Comment:
It might be easier to read if this variable was shortened to `destinations` (even though they're not Navigation**Destinations**, they're how the app defines its destinations.